src/shared/watchdog.c

```diff
@@ -114,9 +117,37 @@ int watchdog_set_timeout(usec_t *usec) {
         return r;
 }
 
+usec_t watchdog_runtime_wait(void) {
+        usec_t rtwait = USEC_INFINITY;
+        usec_t ntime;
+
+        if (timestamp_is_set(watchdog_timeout)) {
```

Comment:
nitpick: in the interest of keeping indentation levels low, I'd invert the condition and do an early exit. i.e.:

```c
if (!timestamp_is_set(watchdog_timeout))
        return USEC_INFINITY;
```